core/src/providers/embedder.rs

```diff
@@ -26,6 +26,8 @@ pub trait Embedder {
     async fn encode(&self, text: &str) -> Result<Vec<usize>>;
     async fn decode(&self, tokens: Vec<usize>) -> Result<String>;
 
+    async fn custom_tokenize(&self, text: String) -> Result<(Vec<usize>, Vec<String>)>;
```

Comment:
(I decided to prefix by `custom` to make it clear that's not from tiktoken but custom for us.)

Reply:
Well then I would push the custom into tiktoken :)

But really I think we can call it `tokenize` everywhere and add a small comment in tiktoken around our own functions

Reply:
Looking